File: src/test/java/com/jesperancinha/service/services/QueryAirportServiceIT.java
package com.jesperancinha.service.services;

import com.jesperancinha.service.configuration.AirportsAppConfiguration;
import com.jesperancinha.service.pojos.Airport;
import org.apache.camel.test.spring.CamelSpringDelegatingTestContextLoader;
import org.apache.camel.test.spring.CamelSpringJUnit4ClassRunner;
import org.apache.camel.test.spring.CamelTestContextBootstrapper;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.annotation.DirtiesContext;
import org.springframework.test.context.BootstrapWith;
import org.springframework.test.context.ContextConfiguration;

import java.util.List;

import static org.junit.Assert.assertNotNull;

@RunWith(CamelSpringJUnit4ClassRunner.class)
@BootstrapWith(CamelTestContextBootstrapper.class)
@ContextConfiguration(classes = {
        AirportsAppConfiguration.class
},
        loader = CamelSpringDelegatingTestContextLoader.class)
@DirtiesContext
public class QueryAirportServiceIT {

    @Autowired
    QueryAirportService queryAirportService;

    @Test
    public void getAirportsByCountryName() throws Exception {
        List<Airport> airports = queryAirportService.getAirportsByCountryName("Portugal");
        assertNotNull(queryAirportService);
        assertNotNull(airports);
    }

    @Test
    public void getAirportsByCountryCode() throws Exception {

    }

}

File: airports-app/src/test/kotlin/org/jesperancinha/airports/QueryAirportServiceIT.kt
package org.jesperancinha.airports

import io.kotest.matchers.collections.shouldNotBeEmpty
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import org.apache.camel.test.spring.junit5.CamelSpringBootTest
import org.jesperancinha.airports.pojos.Airport
import org.jesperancinha.airports.services.QueryAirportService
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.DirtiesContext

@CamelSpringBootTest
@EnableAutoConfiguration
@SpringBootTest
@DirtiesContext
class QueryAirportServiceIT @Autowired constructor(
    private val queryAirportService: QueryAirportService
) {

    @Test
    fun `should get airports by name`() {
        queryAirportService.shouldNotBeNull()
        val airports: List<Airport> = queryAirportService.getAirportsByCountryName("Portugal")
        airports.shouldNotBeNull()
    }

    @Test
    fun `should get airports by country code`() {
        val airports: List<Airport> = queryAirportService.getAirportsByCountryCode("PT")

        airports.shouldNotBeEmpty()
        airports.forEach { airport -> airport.isoCountry shouldBe "PT" }
    }
}